engine/table/src/main/java/io/deephaven/engine/table/impl/ZeroKeyUpdateByManager.java

```diff
@@ -0,0 +1,102 @@
+package io.deephaven.engine.table.impl;
+
+import io.deephaven.api.updateby.UpdateByControl;
+import io.deephaven.engine.rowset.*;
+import io.deephaven.engine.table.*;
+import io.deephaven.engine.table.impl.updateby.UpdateByWindow;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.LinkedList;
+import java.util.Map;
+
+public class ZeroKeyUpdateByManager extends UpdateBy {
+    /** The output table for this UpdateBy operation */
+    final QueryTable result;
+
+    // this manager has only one bucket, managed by this object
+    final UpdateByBucketHelper zeroKeyUpdateBy;
+
+    /**
+     * Perform an updateBy without any key columns.
+     *
+     * @param description the operation description
+     * @param operators the operations to perform
+     * @param windows the unique windows for this UpdateBy
+     * @param inputSources the primitive input sources
+     * @param operatorInputSourceSlots maps the operators to source indices
+     * @param source the source table
+     * @param resultSources the result sources
+     * @param timestampColumnName the column to use for all time-aware operators
+     * @param redirHelper the row redirection helper for dense output sources
+     * @param control the control object.
+     */
+    protected ZeroKeyUpdateByManager(
+            @NotNull final String description,
+            @NotNull UpdateByOperator[] operators,
```

Comment:
Could use some more consistency here, regarding things like `final`.

Reply:
Could use guidance on `final` for function parameters.